backend/internal/github/sharedclient/sharedclient.go

```diff
@@ -323,3 +323,70 @@ func (api *CommonAPI) GetTeamMembers(ctx context.Context, teamID int64) ([]*gith
 	members, _, err := api.Client.Teams.ListTeamMembers(ctx, teamID, nil)
 	return members, err
 }
+
+func (api *CommonAPI) CreateEmptyCommit(ctx context.Context, owner, repo string) error {
+	ghRepo, err := api.GetRepository(ctx, owner, repo)
+	if err != nil {
+		return err
+	}
+	if ghRepo.DefaultBranch == nil {
+		return errs.MissingDefaultBranchError()
+	}
+
+	// Get the reference to main branch
+	ref, _, err := api.Client.Git.GetRef(context.Background(), owner, repo, "heads/"+*ghRepo.DefaultBranch)
+	if err != nil {
+		return err
+	}
+
+	// if no parent commit exists, refer to the empty tree
+	tree := "4b825dc642cb6eb9a060e54bf8d69288fbee4904"
+	var parentCommitSHA string
+	if ref.Object != nil {
+		// Get the commit from the ref
+		parentCommitSHA = ref.Object.GetSHA()
+		if parentCommitSHA != "" {
```

Comment:
This null check is ineffective because GetSHA returns 0 for a missing SHA. Additionally, I am not sure if we want to do nothing if the commit does not exist. I would get rid of this check and let the `GetCommit()` error if the SHA is malformatted.